tests/unit/parsers/test.manifestjson.js

```diff
@@ -192,24 +192,36 @@ describe('ManifestJSONParser', () => {
   });
 
   describe('bad permissions', () => {
-    it('should not error if permission is a string (even if unknown)', () => {
-      const addonLinter = new Linter({ _: ['bar'] });
-      const json = validManifestJSON({
-        permissions: ['identity', 'fileSystem'],
-      });
-
-      const manifestJSONParser = new ManifestJSONParser(
-        json,
-        addonLinter.collector
-      );
-      expect(manifestJSONParser.isValid).toEqual(false);
-      const { warnings } = addonLinter.collector;
-      expect(addonLinter.collector.errors.length).toBe(0);
-      assertHasMatchingError(warnings, {
-        code: messages.MANIFEST_PERMISSIONS.code,
-        message: /Unknown permissions "fileSystem"/,
+    for (const manifestKey of ['permissions', 'optional_permissions']) {
+      const expectedMsgCode =
+        manifestKey === 'permissions'
+          ? messages.MANIFEST_PERMISSIONS.code
+          : messages.MANIFEST_OPTIONAL_PERMISSIONS.code;
+      const json =
+        manifestKey === 'permissions'
+          ? validManifestJSON({
+              permissions: ['identity', 'fileSystem'],
+            })
+          : validManifestJSON({
+              optional_permissions: ['idle', 'fileSystem'],
+              applications: { gecko: { strict_max_version: '55.0' } },
+            });
+      it(`should not error if ${manifestKey} is a string (even if unknown)`, () => {
+        const addonLinter = new Linter({ _: ['bar'] });
+        const manifestJSONParser = new ManifestJSONParser(
+          json,
+          addonLinter.collector
+        );
+        expect(manifestJSONParser.isValid).toEqual(false);
+        const { warnings } = addonLinter.collector;
+        console.log({ warnings });
```

Comment:
I think somewhere around here the test is failing, I tried debugging but couldn't find much
here's the log btw

● ManifestJSONParser › bad permissions › should not error if permissions is a string (even if unknown)

    expect(received).toBe(expected) // Object.is equality

    Expected: true

    Received: false

      292 |   }; // eslint-disable-line

      293 | 

    > 294 |   let result;

          |       ^

      295 | 

      296 |   function reset() {

      297 |     process.exit = _exit;

      at assertHasMatchingError (tests/unit/helpers.js:294:7)

      at Object.<anonymous> (tests/unit/parsers/test.manifestjson.js:234:45)

  ● ManifestJSONParser › bad permissions › should not error if optional_permissions is a string (even if unknown)

    expect(received).toBe(expected) // Object.is equality

    Expected: true

    Received: false

      292 |   }; // eslint-disable-line

      293 | 

    > 294 |   let result;

          |       ^

      295 | 

      296 |   function reset() {

      297 |     process.exit = _exit;

      at assertHasMatchingError (tests/unit/helpers.js:294:7)

      at Object.<anonymous> (tests/unit/parsers/test.manifestjson.js:234:45)

Reply:
This test is failing in the `assertHasMatchingError` helpers, in particular when it does assert that the linting message is matching the one at line 221:

in the previous version of this test case message was a RegExp, in this new version is a string and so the helper would expect the linting message to be strictly equal to this string.

What you want here is to generate a RegExp for the interpolated string:

```js
         assertHasMatchingError(warnings, {
           code: expectedMsgCode,
           message: new RegExp(`Unknown ${manifestKey} "fileSystem"`),
         });
```

This should make the test to pass again as expected, for both permissions and optional_permissions.